### ISSUE TITLE
support macOS?

### ISSUE THREAD

**Issue by danipen** [teobugslayer/AvaloniaCoreRTDemo#11]
Hi, I tried to compile and run this sample in macOS.

```
$ AvaloniaCoreRTDemo$ dotnet publish -r osx-x64 -c release
Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj (in 1.79 sec).
  GenerateAvaloniaResourcesTask -> Root: /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src, 4 resources, Output:obj/Release/net5.0/osx-x64//Avalonia/resources
  avares -> name:/AboutWindow.xaml, path: /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AboutWindow.xaml, size:6946, ItemSpec:AboutWindow.xaml
  avares -> name:/App.xaml, path: /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/App.xaml, size:405, ItemSpec:App.xaml
  avares -> name:/MainWindow.xaml, path: /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/MainWindow.xaml, size:1550, ItemSpec:MainWindow.xaml
  avares -> name:/app.ico, path: /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/app.ico, size:183198, ItemSpec:app.ico
  AvaloniaCoreRTDemo -> /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/bin/Release/net5.0/osx-x64/AvaloniaCoreRTDemo.dll
  Generating native code
/Users/danipen/.nuget/packages/avalonia/0.10.0/lib/netcoreapp2.0/Avalonia.Base.dll : warning IL9702: Assembly 'Avalonia.Base' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/avalonia/0.10.0/lib/netcoreapp2.0/Avalonia.OpenGL.dll : warning IL9702: Assembly 'Avalonia.OpenGL' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/avalonia/0.10.0/lib/netcoreapp2.0/Avalonia.MicroCom.dll : warning IL9702: Assembly 'Avalonia.MicroCom' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/runtime.osx-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.5.21230.2/framework/System.Linq.Expressions.dll : warning IL9702: Assembly 'System.Linq.Expressions' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/runtime.osx-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.5.21230.2/framework/System.ComponentModel.TypeConverter.dll : warning IL9702: Assembly 'System.ComponentModel.TypeConverter' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/runtime.osx-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.5.21230.2/sdk/System.Private.CoreLib.dll : warning IL9702: Assembly 'System.Private.CoreLib' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/tmds.dbus/0.9.0/lib/netstandard2.0/Tmds.DBus.dll : warning IL9702: Assembly 'Tmds.DBus' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/runtime.osx-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.5.21230.2/framework/System.Diagnostics.DiagnosticSource.dll : warning IL9702: Assembly 'System.Diagnostics.DiagnosticSource' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
/Users/danipen/.nuget/packages/runtime.osx-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.5.21230.2/framework/System.Formats.Asn1.dll : warning IL9702: Assembly 'System.Formats.Asn1' produced AOT analysis warnings. [/Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/AvaloniaCoreRTDemo.csproj]
  ld: warning: could not create compact unwind for _System_Net_Http_System_Net_Http_Headers_QPackStaticTable___cctor: stack subq instruction is too different from dwarf stack size
  ld: warning: could not create compact unwind for _Avalonia_Themes_Default_CompiledAvaloniaXaml__EmbeddedResource__Populate_Avalonia_Themes_Default_ToggleSwitch_xaml: stack subq instruction is too different from dwarf stack size
  ld: warning: could not create compact unwind for _AvaloniaCoreRTDemo_AvaloniaCoreRTDemo_AboutWindow___XamlIlPopulate: stack subq instruction is too different from dwarf stack size
  ld: warning: could not create compact unwind for _S_P_CoreLib_System_Number__Dragon4: stack subq instruction is too different from dwarf stack size
  ld: warning: can't parse dwarf compilation unit info in obj/Release/net5.0/osx-x64/native/AvaloniaCoreRTDemo.o
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_Backtrace' hides a non-existent symbol '__Unwind_Backtrace'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_Backtrace' hides a non-existent symbol '__Unwind_Backtrace'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_DeleteException' hides a non-existent symbol '__Unwind_DeleteException'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_DeleteException' hides a non-existent symbol '__Unwind_DeleteException'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_FindEnclosingFunction' hides a non-existent symbol '__Unwind_FindEnclosingFunction'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_FindEnclosingFunction' hides a non-existent symbol '__Unwind_FindEnclosingFunction'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_Find_FDE' hides a non-existent symbol '__Unwind_Find_FDE'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_Find_FDE' hides a non-existent symbol '__Unwind_Find_FDE'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_ForcedUnwind' hides a non-existent symbol '__Unwind_ForcedUnwind'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_ForcedUnwind' hides a non-existent symbol '__Unwind_ForcedUnwind'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetCFA' hides a non-existent symbol '__Unwind_GetCFA'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetCFA' hides a non-existent symbol '__Unwind_GetCFA'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetDataRelBase' hides a non-existent symbol '__Unwind_GetDataRelBase'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetDataRelBase' hides a non-existent symbol '__Unwind_GetDataRelBase'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetGR' hides a non-existent symbol '__Unwind_GetGR'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetGR' hides a non-existent symbol '__Unwind_GetGR'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetIP' hides a non-existent symbol '__Unwind_GetIP'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetIP' hides a non-existent symbol '__Unwind_GetIP'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetIPInfo' hides a non-existent symbol '__Unwind_GetIPInfo'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetIPInfo' hides a non-existent symbol '__Unwind_GetIPInfo'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetLanguageSpecificData' hides a non-existent symbol '__Unwind_GetLanguageSpecificData'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetLanguageSpecificData' hides a non-existent symbol '__Unwind_GetLanguageSpecificData'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetRegionStart' hides a non-existent symbol '__Unwind_GetRegionStart'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetRegionStart' hides a non-existent symbol '__Unwind_GetRegionStart'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_GetTextRelBase' hides a non-existent symbol '__Unwind_GetTextRelBase'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_GetTextRelBase' hides a non-existent symbol '__Unwind_GetTextRelBase'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_RaiseException' hides a non-existent symbol '__Unwind_RaiseException'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_RaiseException' hides a non-existent symbol '__Unwind_RaiseException'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_Resume' hides a non-existent symbol '__Unwind_Resume'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_Resume' hides a non-existent symbol '__Unwind_Resume'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_Resume_or_Rethrow' hides a non-existent symbol '__Unwind_Resume_or_Rethrow'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_Resume_or_Rethrow' hides a non-existent symbol '__Unwind_Resume_or_Rethrow'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_SetGR' hides a non-existent symbol '__Unwind_SetGR'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_SetGR' hides a non-existent symbol '__Unwind_SetGR'
  ld: warning: linker symbol '$ld$hide$os10.4$__Unwind_SetIP' hides a non-existent symbol '__Unwind_SetIP'
  ld: warning: linker symbol '$ld$hide$os10.5$__Unwind_SetIP' hides a non-existent symbol '__Unwind_SetIP'
  ld: warning: linker symbol '$ld$hide$os10.4$___deregister_frame' hides a non-existent symbol '___deregister_frame'
  ld: warning: linker symbol '$ld$hide$os10.5$___deregister_frame' hides a non-existent symbol '___deregister_frame'
  ld: warning: linker symbol '$ld$hide$os10.4$___deregister_frame_info' hides a non-existent symbol '___deregister_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.5$___deregister_frame_info' hides a non-existent symbol '___deregister_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.6$___deregister_frame_info' hides a non-existent symbol '___deregister_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.4$___deregister_frame_info_bases' hides a non-existent symbol '___deregister_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.5$___deregister_frame_info_bases' hides a non-existent symbol '___deregister_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.6$___deregister_frame_info_bases' hides a non-existent symbol '___deregister_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame' hides a non-existent symbol '___register_frame'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame' hides a non-existent symbol '___register_frame'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame_info' hides a non-existent symbol '___register_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame_info' hides a non-existent symbol '___register_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.6$___register_frame_info' hides a non-existent symbol '___register_frame_info'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame_info_bases' hides a non-existent symbol '___register_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame_info_bases' hides a non-existent symbol '___register_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.6$___register_frame_info_bases' hides a non-existent symbol '___register_frame_info_bases'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame_info_table' hides a non-existent symbol '___register_frame_info_table'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame_info_table' hides a non-existent symbol '___register_frame_info_table'
  ld: warning: linker symbol '$ld$hide$os10.6$___register_frame_info_table' hides a non-existent symbol '___register_frame_info_table'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame_info_table_bases' hides a non-existent symbol '___register_frame_info_table_bases'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame_info_table_bases' hides a non-existent symbol '___register_frame_info_table_bases'
  ld: warning: linker symbol '$ld$hide$os10.6$___register_frame_info_table_bases' hides a non-existent symbol '___register_frame_info_table_bases'
  ld: warning: linker symbol '$ld$hide$os10.4$___register_frame_table' hides a non-existent symbol '___register_frame_table'
  ld: warning: linker symbol '$ld$hide$os10.5$___register_frame_table' hides a non-existent symbol '___register_frame_table'
  ld: warning: linker symbol '$ld$hide$os10.6$___register_frame_table' hides a non-existent symbol '___register_frame_table'
  AvaloniaCoreRTDemo -> /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/bin/Release/net5.0/osx-x64/publish/
```

It compiles and generates the executable, but when I try to run it, I the process is just get aborted:

```
AvaloniaCoreRTDemo$ /Users/danipen/Documents/GitHub/AvaloniaCoreRTDemo/src/bin/Release/net5.0/osx-x64/publish/AvaloniaCoreRTDemo
[1]    23921 abort
```

How can I get some information about the crash? Any help would be appreciated.

**Comment by danipen** [comment]
Also reported in https://github.com/dotnet/runtimelab/issues/1053

**Comment by teobugslayer** [comment]
Unfortunately I no longer have access to a macOS machine so I am unable to provide assistance.

**Comment by josephmoresena** [comment]
Trying it into a VM:
`
Unhandled Exception: System.Collections.Generic.KeyNotFoundException: The given key 'Avalonia.Native.Interop.IAvaloniaNativeFactory' was not present in the dictionary.
   at System.Collections.Concurrent.ConcurrentDictionary`2.ThrowKeyNotFoundException(TKey) + 0x54
   at System.Collections.Concurrent.ConcurrentDictionary`2.get_Item(TKey) + 0x50
   at Avalonia.MicroCom.MicroComRuntime.CreateProxyFor[T](IntPtr, Boolean) + 0x41
   at Avalonia.Native.AvaloniaNativePlatform.Initialize(IntPtr, AvaloniaNativePlatformOptions) + 0x35
   at Avalonia.Native.AvaloniaNativePlatform.Initialize(AvaloniaNativePlatformOptions) + 0xcb
   at Avalonia.AvaloniaNativePlatformExtensions.<>c__DisplayClass0_0`1.<UseAvaloniaNative>b__0() + 0x69
   at Avalonia.Controls.AppBuilderBase`1.Setup() + 0xcb
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime[T](T, String[], ShutdownMode) + 0x53
   at AvaloniaCoreRTDemo!<BaseAddress>+0xf285e7
zsh: abort      
`

**Comment by teobugslayer** [comment]
Closed after merging #13
